src/core/services/compiler/compiler.js

```diff
@@ -6,7 +6,20 @@
  */
 angular
   .module('material.core')
-  .service('$mdCompiler', MdCompilerService);
+  .service('$mdCompiler', MdCompilerService)
+  .config(configureCompiler);
+
+/**
+ * Function that is called on the configuration phase of AngularJS.
+ * It is used to determine if bindings should be pre-assigned or not.
+ */
+function configureCompiler($compileProvider, $provide) {
+  $provide.constant('$$mdPreAssignBindings',
+    // For Angular versions that do not support `preAssignBindingsEnabled`, always pre-assign the
+    // bindings.
+    $compileProvider.preAssignBindingsEnabled ? $compileProvider.preAssignBindingsEnabled() : true
```

Comment:
@gkalpak your implementation is good, I'd just drop the ` ? true : false` part as it's redundant here.